game_management/game_input_handler.py

```diff
@@ -0,0 +1,593 @@
+            # If we're not on a menu, pan the map when you press down.
+            # Holding Ctrl will pan the map 5 spaces.
+            if is_ctrl_key:
+                game_state.map_pos = game_state.map_pos[0], clamp(game_state.map_pos[1] + 5, -1, 69)
+            else:
+                game_state.map_pos = game_state.map_pos[0], clamp(game_state.map_pos[1] + 1, -1, 69)
+
+
+def on_key_arrow_up(game_controller: GameController, game_state: GameState, is_ctrl_key: bool):
+    """
+    Handles an Arrow Key up event in the game loop.
+    :param game_controller: The current GameController object.
+    :param game_state: The current GameState object.
+    :param is_ctrl_key: Whether the CTRL key has also been pressed.
+    """
+    if game_state.on_menu:
+        game_controller.menu.navigate(up=True)
+    elif game_state.game_started:
+        if game_state.board.overlay.is_constructing():
+            game_state.board.overlay.navigate_constructions(down=False)
+        elif game_state.board.overlay.is_blessing():
+            game_state.board.overlay.navigate_blessings(down=False)
+        elif game_state.board.overlay.is_setl_click():
+            game_state.board.overlay.navigate_setl_click(up=True)
+        elif game_state.board.overlay.is_controls():
+            game_state.board.overlay.show_additional_controls = False
+        elif game_state.board.overlay.is_pause():
+            game_state.board.overlay.navigate_pause(down=False)
+        elif game_state.board.overlay.is_standard():
+            game_state.board.overlay.navigate_standard(down=False)
+        else:
+            game_state.board.overlay.remove_warning_if_possible()
+            # If we're not on a menu, pan the map when you press up.
+            # Holding Ctrl will pan the map 5 spaces.
+            if is_ctrl_key:
+                game_state.map_pos = game_state.map_pos[0], clamp(game_state.map_pos[1] - 5, -1, 69)
+            else:
+                game_state.map_pos = game_state.map_pos[0], clamp(game_state.map_pos[1] - 1, -1, 69)
+
+
+def on_key_arrow_left(game_controller: GameController, game_state: GameState, is_ctrl_key: bool):
+    """
+    Handles an Arrow Key left event in the game loop.
+    :param game_controller: The current GameController object.
+    :param game_state: The current GameState object.
+    :param is_ctrl_key: Whether the CTRL key has also been pressed.
+    """
+    if game_state.on_menu:
+        game_controller.menu.navigate(left=True)
+    if game_state.game_started and game_state.board.overlay.is_constructing():
+        if game_state.board.overlay.current_construction_menu is ConstructionMenu.PROJECTS and \
+                len(game_state.board.overlay.available_constructions) > 0:
+            game_state.board.overlay.current_construction_menu = ConstructionMenu.IMPROVEMENTS
+            game_state.board.overlay.selected_construction = game_state.board.overlay.available_constructions[
+                0]
```

Comment:
Probably better to put this guy back on the previous line

Reply:
Whoa, weird that this even happened at all :O

Reply:
[588b895](https://github.com/ChrisNeedham24/microcosm/pull/111/commits/588b895112c545f17b9883b616153d3514b540f7)